handler.go

```diff
@@ -954,6 +956,19 @@ func (h *Handler) writeJSONQueryResponse(w http.ResponseWriter, resp *QueryRespo
 	return json.NewEncoder(w).Encode(resp)
 }
 
+func (h *Handler) writeJSONResponse(w http.ResponseWriter, thing interface{}) {
```

Comment:
Is there a reason you did not want to use the existing:
func (resp *QueryResponse) MarshalJSON() ([]byte, error) 
Instead of your custom func (h *Handler) writeJSONResponse?

It seems like it would be better to keep a consistent result structure.